File: src/types/patients.ts
export interface PatientListItem {
  patient_id: string;
  patient_code: string;
  full_name: string;
  primary_doctor_id?: string | null;
  liver_score?: number | null;
  risk_score?: number | null;
  latest_fibroscan_kpa?: number | null;
  sgpt?: number | null;
  active_package_name?: string | null;
  score_calculated_at?: string | null;
  bmi?: number | null;
}

/** @deprecated Use PatientListItem — kept for list store compatibility */
export interface Patient {
  id: string;
  patientCode: string;
  fullName: string;
  email?: string;
  phone?: string;
  status: 'active' | 'inactive' | 'pending';
  assignedDoctor?: string;
  lastVisit?: string;
  bmi?: number | null;
  riskScore?: number | null;
  liverScore?: number | null;
}

export interface PatientSummaryCard {
  patientCode: string;
  name: string;
  ageGender: string;
  bmi: number | null;
  riskCategory: string;
  diagnosis: string;
  diabetes: string;
  alcohol: string;
  latestLiverFibrosisScanKpa: number | null;
  latestAlt: number | null;
  currentPlan: string | null;
  liverScore: number | null;
  fibrosisStage: string | null;
  alerts: string[];
  journeyStatus: string;
}

export interface PatientDetail {
  patient: Record<string, unknown>;
  dashboard: Record<string, unknown> | null;
  summaryCard: PatientSummaryCard;
  addresses: Record<string, unknown>[];
  allergyAlerts: Record<string, unknown>[];
}

export interface PatientTrendPoint {
  patient_id: string;
  snapshot_date: string;
  weight_kg?: number | null;
  bmi?: number | null;
  sgpt?: number | null;
  sgot?: number | null;
  hba1c?: number | null;
  triglycerides?: number | null;
  liverFibrosisScanKpa?: number | null;
  cap_dbm?: number | null;
  liver_score?: number | null;
  compliance_score?: number | null;
}

export interface PatientDashboardKpis {
  liverScore: number | null;
  riskScore: number | null;
  complianceScore: number | null;
  bmi: number | null;
  weightKg: number | null;
  heightCm: number | null;
  latestLiverFibrosisScanKpa: number | null;
  latestCapDbm: number | null;
  fibrosisStage: string | null;
  steatosisGrade: string | null;
  sgpt: number | null;
  sgot: number | null;
  hba1c: number | null;
  triglycerides: number | null;
  activePackage: string | null;
  packageStart: string | null;
  packageEnd: string | null;
  dietCompliance: number | null;
  exerciseCompliance: number | null;
  medicineCompliance: number | null;
  scoreCalculatedAt: string | null;
  homeVisitsTotal: number;
  homeVisitsCompleted: number;
  prescriptionsTotal: number;
  prescriptionsApproved: number;
}

export interface PatientCompliancePoint {
  checkinWeekStart: string;
  weightKg: number | null;
  dietCompliancePercent: number | null;
  exerciseCompliancePercent: number | null;
  medicineCompliancePercent: number | null;
  alcoholIntake: string;
  submittedAt: string;
}

export interface PatientDashboardData {
  kpis: PatientDashboardKpis;
  trends: PatientTrendPoint[];
  compliance: PatientCompliancePoint[];
}

export interface TimelineEvent {
  id?: string;
  occurred_at: string;
  activity_type: string;
  role?: string | null;
  description: string;
  event_category?: string;
  status?: string | null;
  source?: string;
}

export interface PatientHistory {
  conditions: Record<string, unknown>[];
  liverHistory: Record<string, unknown> | null;
  medications: Record<string, unknown>[];
  allergies: Record<string, unknown>[];
  surgeries: Record<string, unknown>[];
  vaccinations: Record<string, unknown>[];
  familyMembers: Record<string, unknown>[];
  defaultConditionCodes: { code: string; name: string }[];
}

export interface DashboardOverview {
  stats: {
    activePatients: number;
    visitsToday: number;
    pendingPrescriptions: number;
    highRiskPatients: number;
  };
  charts: {
    patientsByCity: { city_name: string; patients: number }[];
    clinicTrends: {
      snapshot_date: string;
      avg_bmi: number | null;
      avg_alt: number | null;
      avg_liver_fibrosis_scan: number | null;
    }[];
  };
}

export interface PatientFilters extends Record<string, unknown> {
  status: '' | 'active' | 'inactive' | 'pending';
  assignedDoctor: string;
}

export const DEFAULT_PATIENT_FILTERS: PatientFilters = {
  status: '',
  assignedDoctor: '',
};

export function mapListItemToPatient(row: PatientListItem): Patient {
  return {
    id: row.patient_id,
    patientCode: row.patient_code,
    fullName: row.full_name,
    status: 'active',
    bmi: row.bmi,
    riskScore: row.risk_score,
    liverScore: row.liver_score,
    lastVisit: row.score_calculated_at ?? undefined,
  };
}

File: apps/api/app/schemas/patients.py
from __future__ import annotations

from typing import Any
from uuid import UUID

from pydantic import Field

from app.schemas.common import BaseSchema


class PatientListItem(BaseSchema):
    patient_id: UUID = Field(alias="patientId")
    patient_code: str = Field(alias="patientCode")
    full_name: str = Field(alias="fullName")
    mobile: str | None = None
    bmi: float | None = None
    risk_score: float | None = Field(default=None, alias="riskScore")
    liver_score: float | None = Field(default=None, alias="liverScore")
    journey_status: str | None = Field(default=None, alias="journeyStatus")
    primary_doctor_name: str | None = Field(default=None, alias="primaryDoctorName")


class PaginatedPatients(BaseSchema):
    items: list[PatientListItem]
    total: int
    page: int
    page_size: int = Field(alias="pageSize")
    total_pages: int = Field(alias="totalPages")


class PatientSummaryCard(BaseSchema):
    patient_code: str = Field(alias="patientCode")
    name: str
    age_gender: str = Field(alias="ageGender")
    bmi: float | None = None
    risk_category: str = Field(alias="riskCategory")
    diagnosis: str | None = None
    diabetes: str | None = None
    alcohol: str | None = None
    current_plan: str | None = Field(default=None, alias="currentPlan")
    liver_score: float | None = Field(default=None, alias="liverScore")
    fibrosis_stage: str | None = Field(default=None, alias="fibrosisStage")
    alerts: list[str] = Field(default_factory=list)
    journey_status: str | None = Field(default=None, alias="journeyStatus")


class PatientDetail(BaseSchema):
    patient: dict[str, Any]
    dashboard: dict[str, Any] | None = None
    summary_card: PatientSummaryCard = Field(alias="summaryCard")
    addresses: list[dict[str, Any]] = Field(default_factory=list)
    allergy_alerts: list[dict[str, Any]] = Field(default_factory=list, alias="allergyAlerts")


class PatientClinicalContext(BaseSchema):
    orders: list[dict[str, Any]]
    payments: list[dict[str, Any]]
    pathology_reports: list[dict[str, Any]] = Field(default_factory=list, alias="pathologyReports")
    scans: list[dict[str, Any]] = Field(default_factory=list)
    appointments: list[dict[str, Any]] = Field(default_factory=list)
    reports: list[dict[str, Any]] = Field(default_factory=list)
